Contracts/Presale/PresaleContract.sol

```diff
@@ -137,7 +137,21 @@ contract PresaleContract is WithdrawContract {
         return usersBalance[_user].tokens;
     }
     function changeBalance(address _user, uint _balance) public onlyOwner() {
+        require(_user != address(0));
         Balance storage balance = usersBalance[_user];
+        require(soldTokensAmount >= balance.tokens);
+        require(balance.tokens != _balance);
+        
+        if (balance.tokens == 0 && balance.ethers == 0) {
+            investors.push(_user);
+        }
+        
+        if (balance.tokens > _balance) {
+            soldTokensAmount -= (balance.tokens - _balance);

```

Comment:
@Yaschenko if balance.tokens > _balance - this means that we are removing tokens from the user - this means the total sold tokens is now MORE, no?